squash-honeycomb.ipynb

```diff
@@ -0,0 +1,381 @@
+    "- A detection of particular astronomical object \n",
+    "\n",
+    "**Datasets** are used to partition your data into separate and queryable sets. See [best practices for defining datasets](https://docs.honeycomb.io/getting-data-in/datasets/best-practices/) in Honeycomb. \n",
+    "\n",
+    "Let's start by identifying a `validate_drp` run as our event and creating a new dataset for the metrics measured by each run. \n",
+    "\n",
+    "For illustration purposes we'll collect those metrics from the SQuaSH API:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "SQUASH_API_URL = \"https://squash-restful-api-demo.lsst.codes/\""
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import requests\n",
+    "r = requests.get(SQUASH_API_URL + \"/job/1\").json()\n",
+    "    \n",
+    "    \n",
+    "data = {'id': r['id'],\n",
+    "        'date_created': r['date_created'],\n",
+    "        'filter_name': r['meta']['filter_name'],\n",
+    "        'dataset': r['ci_dataset']}\n",
+    "            \n",
+    "# events should be flat dict\n",
+    "for meas in r['measurements']:\n",
+    "    data[meas['metric']] = meas['value']\n",
+    "    \n",
+    "data"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "This looks like an structured log entry, it also looks like a property set or it may recall something else..."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Sending events to Honeycomb\n",
+    "\n",
+    "1. When honeycomb gets a request to add an event to a dataset that doesn't exist yet, it creates the dataset.\n",
+    "2. The dataset schema is inferred automatically. It also will infer automatically if you add new fields to your events.\n",
+    "\n",
+    "3. Events are added to the dataset for querying.\n",
+    "\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "NOTE: In order to execute this cell you will need the HONEY_API_KEY. It is avaibale from \"Team Settings\" in the UI once you join the `lsst-square` team."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import libhoney\n",
+    "\n",
+    "HONEY_API_KEY = \"\"  # Obtained from the UI under \"Team Settings\"\n",
+    "\n",
+    "libhoney.init(writekey=HONEY_API_KEY, dataset=\"single-event-demo\")\n",
+    "libhoney.send_now(data)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "That's it! Check the new dataset at https://ui.honeycomb.io/lsst-square/datasets"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "NOTE: this demo dataset may already exist from previous executions of this notebook. "
```

Comment:
What happens if it already exists?  Do I get a 403, or something?

Reply:
Rephrased like this:
NOTE: this demo dataset may already exist from previous executions of this notebook, make sure you delete it before running the notebook if you want to start from an empty dataset.

Reply:
It still leaves me wondering "What happens if I insert into a dataset that already exists?"  Do you get an error, or duplicate entries, or does it ignore duplicates?

Reply:
Oh, great question. It uses the Timestamp as key, so it wont duplicate events if they have the same timestamp.

Reply:
NOTE: this demo dataset may already exist from previous executions of this notebook, make sure you delete it before running the notebook if you want to start from an empty dataset. If you don't, note that as long as events have different timestamps they will be inserted in the existing dataset.